### PR TITLE
SONARJS-971 NPE in SuperInvocationCheck

### PR DESCRIPTION
**Pull request by inverno**
constructor inside object literal

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/SuperInvocationCheck.java [hunk 1]
```diff
@@ -284,7 +284,12 @@ private static boolean isConstructor(FunctionTree tree) {
   }
 
   private static Optional<ClassTree> getEnclosingClass(Tree tree) {
```
Comment (ivandalbosco) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/SuperInvocationCheck.java]: @inverno How about renaming this method? Into getEnclosingType()?

File: javascript-checks/src/test/resources/checks/superInvocation.js [hunk 2]
```diff
@@ -295,7 +295,35 @@ class B40 extends A40 {
...
+class B41 extends A40 {
+
   constructor() {
+    super();
+  }
+
+  createObject() {
+    var obj = {
+      constructor() {       // OK
+      }
```
Comment (ivandalbosco) [inline javascript-checks/src/test/resources/checks/superInvocation.js]: @inverno I would add a noncompliant case. With a compliant case, we don't know if the the case is really compliant or simply ignored. Suggestion: add a line with a call to super().

Comment (ivandalbosco) [inline javascript-checks/src/test/resources/checks/superInvocation.js]: @inverno How about adding a test where the constructor is defined separately, e.g.,
```
createObject2() {
    var obj = { };
    obj.constructor = function() {
      super();      // Noncompliant ...
    };
    return obj;
  }

createObject3() {
    var obj = { };
    obj.constructor = function() {
      // OK
    };
    return obj;
  }
```
OK, I know, people already told me I add too many tests...

File: javascript-checks/src/main/java/org/sonar/javascript/checks/SuperInvocationCheck.java (hunk 1 above)
Comment (inverno) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/SuperInvocationCheck.java]: Done

File: javascript-checks/src/test/resources/checks/superInvocation.js (hunk 2 above)
Comment (inverno) [inline javascript-checks/src/test/resources/checks/superInvocation.js]: It's a very good idea, thanks. Done

Comment (inverno) [inline javascript-checks/src/test/resources/checks/superInvocation.js]: I did already use `super()` in the `top_level_object` test. Now I also do it in the new cases you suggested below. I think it should be enough.

**Pull request merged**